diff --git a/node_modules/@stencil/core/internal/client/index.js b/node_modules/@stencil/core/internal/client/index.js
index 7ca390b..e10ed7a 100644
--- a/node_modules/@stencil/core/internal/client/index.js
+++ b/node_modules/@stencil/core/internal/client/index.js
@@ -3256,6 +3256,14 @@ var updateComponent = async (hostRef, instance, isInitialLoad) => {
   if (BUILD21.isDev) {
     hostRef.$flags$ |= 1024 /* devOnRender */;
   }
+
+  //========= PDS PATCH START
+  if (elm.hasDSR) {
+    elm.shadowRoot.innerHTML = '';
+    delete elm.hasDSR;
+  }
+  //========= PDS PATCH END
+
   if (BUILD21.hydrateServerSide) {
     await callRender(hostRef, instance, elm, isInitialLoad);
   } else {
@@ -4245,6 +4253,13 @@ var bootstrapLazy = (lazyBundles, options = {}) => {
           registerHost(self, cmpMeta);
           if (BUILD28.shadowDom && cmpMeta.$flags$ & 1 /* shadowDomEncapsulation */) {
             if (supportsShadow) {
+
+              //========= PDS PATCH START
+              if (self.shadowRoot) {
+                self.hasDSR = true;
+              }
+              //========= PDS PATCH END
+
               if (!self.shadowRoot) {
                 createShadowRoot.call(self, cmpMeta);
               } else {
