diff --git a/node_modules/@stencil/core/compiler/stencil.js b/node_modules/@stencil/core/compiler/stencil.js
index 1f97bb5..538b44d 100644
--- a/node_modules/@stencil/core/compiler/stencil.js
+++ b/node_modules/@stencil/core/compiler/stencil.js
@@ -251954,7 +251954,7 @@ function createLazyAttachInternalsBinding(cmp) {
                     import_typescript4.default.factory.createIdentifier(HOST_REF_ARG),
                     import_typescript4.default.factory.createIdentifier("$hostElement$")
                   ),
-                  import_typescript4.default.factory.createIdentifier("attachInternals")
+                  import_typescript4.default.factory.createIdentifier("attachInternals?.")
                 ),
                 void 0,
                 []
diff --git a/node_modules/@stencil/core/internal/client/index.js b/node_modules/@stencil/core/internal/client/index.js
index 7ca390b..f450306 100644
--- a/node_modules/@stencil/core/internal/client/index.js
+++ b/node_modules/@stencil/core/internal/client/index.js
@@ -3256,6 +3256,14 @@ var updateComponent = async (hostRef, instance, isInitialLoad) => {
   if (BUILD21.isDev) {
     hostRef.$flags$ |= 1024 /* devOnRender */;
   }
+
+  //========= PDS PATCH START
+  if (elm.hasDSR) {
+    elm.shadowRoot.innerHTML = '';
+    delete elm.hasDSR;
+  }
+  //========= PDS PATCH END
+
   if (BUILD21.hydrateServerSide) {
     await callRender(hostRef, instance, elm, isInitialLoad);
   } else {
@@ -4243,23 +4251,29 @@ var bootstrapLazy = (lazyBundles, options = {}) => {
           super(self);
           self = this;
           registerHost(self, cmpMeta);
-          if (BUILD28.shadowDom && cmpMeta.$flags$ & 1 /* shadowDomEncapsulation */) {
-            if (supportsShadow) {
-              if (!self.shadowRoot) {
-                createShadowRoot.call(self, cmpMeta);
-              } else {
-                if (self.shadowRoot.mode !== "open") {
-                  throw new Error(
-                    `Unable to re-use existing shadow root for ${cmpMeta.$tagName$}! Mode is set to ${self.shadowRoot.mode} but Stencil only supports open shadow roots.`
-                  );
+        }
+        connectedCallback() {
+          //========= PDS PATCH START
+          if (this.hasAttribute('data-ssr')) {
+            this.hasDSR = true;
+          } else {
+            if (BUILD28.shadowDom && cmpMeta.$flags$ & 1 /* shadowDomEncapsulation */) {
+              if (supportsShadow) {
+                if (!this.shadowRoot) {
+                  createShadowRoot.call(this, cmpMeta);
+                } else {
+                  if (this.shadowRoot.mode !== "open") {
+                    throw new Error(
+                      `Unable to re-use existing shadow root for ${cmpMeta.$tagName$}! Mode is set to ${this.shadowRoot.mode} but Stencil only supports open shadow roots.`
+                    );
+                  }
                 }
+              } else if (!BUILD28.hydrateServerSide && !("shadowRoot" in this)) {
+                this.shadowRoot = this;
               }
-            } else if (!BUILD28.hydrateServerSide && !("shadowRoot" in self)) {
-              self.shadowRoot = self;
             }
           }
-        }
-        connectedCallback() {
+          //========= PDS PATCH END
           const hostRef = getHostRef(this);
           if (!hostRef) {
             return;
